src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/FeatureFlagController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Controller\Admin\Configure\AdvancedParameters;
+
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Manages the "Configure > Advanced Parameters > Experimental Features" page.
+ */
+class FeatureFlagController extends FrameworkBundleAdminController
+{
+    public const CONTROLLER_NAME = 'AdminAdvancedParametersFeatureFlags';
+
+    /**
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))", message="Access denied.")
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request): Response
+    {
+        $featureFlagsFormHandler = $this->get('prestashop.admin.feature_flags.form_handler');
+        $featureFlagsForm = $featureFlagsFormHandler->getForm();
+
+        if ($request->isMethod(Request::METHOD_POST)) {
```

Comment:
You can `setMethod` the form to prevent this:
https://symfony.com/doc/4.1/form/action_method.html
https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/Form/NativeRequestHandler.php#L56

Reply:
I am not sure to understand 🤔  . Yes, I can mark this form as POST only in the markup but I will not trust user input?

This controller action handles both the GET and the POST. The GET creates the form and display it, the POST validates form input. So I need to use this `if ($request->isMethod(Request::METHOD_POST))` to find out what I am supposed to do.

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/23703/commits/fa7acd7d832932731006bb05a81e8418aacfa9a7

I cannot use `setMethod()` because the form builder is a generic class but I used default options on the FormType.